walk-in-interview/src/main/java/com/google/job/data/JobPayment.java

```diff
@@ -5,20 +5,39 @@
     private final int min;
     private final int max;
     private final PaymentFrequency paymentFrequency;
+    private final long annualMax;
+
+    private static final int HOURS_PER_YEAR = 8760;
+    /* Note that is an approximate value */
+    private static final int WEEKS_PER_YEAR = 52;
+    private static final int MONTHS_PER_YEAR = 12;
 
     private volatile int hashCode;
 
     // For serialization
     public JobPayment() {
-        this(/* min= */0, /* max= */0, PaymentFrequency.HOURLY);
+        this(/* min= */0, /* max= */0, PaymentFrequency.HOURLY, /* annualMax= */ 0);
     }
 
-    public JobPayment(int min, int max, PaymentFrequency paymentFrequency) {
-        validateParameter(min, max);
+    public JobPayment(int min, int max, PaymentFrequency paymentFrequency, long annualMax) throws IllegalArgumentException {
+        if (min < 0) {
+            throw new IllegalArgumentException("\"min\" should not be negative");
+        }
+
+        if (max < min) {
+            throw new IllegalArgumentException("\"max\" should not be less than \"min\"");
+        }
+
+        long expectedAnnualMax = findExpectedAnnualMax(max, paymentFrequency);
+
+        if (annualMax != expectedAnnualMax) {
+            throw new IllegalArgumentException("annualMax was : " + annualMax + ", but should be: " + expectedAnnualMax);
+        }
 
         this.min = min;
         this.max = max;
         this.paymentFrequency = paymentFrequency;
+        this.annualMax = annualMax;
```

Comment:
i recommend: `this.annualMax = findExpectedAnnualMax(max, paymentFrequency);` instead of using a param. this is essentially equivalent since you are throwing an IllegalArg if the param is not the same as your computation.